medusa/tv/episode.py

```diff
@@ -382,16 +388,35 @@ def location(self):
         """Return the location.
 
         :return:
-        :rtype: location
+        :rtype: text_type
         """
         return self._location
 
     @location.setter
     def location(self, value):
-        log.debug('{id}: Setter sets location to {location}',
-                  {'id': self.series.series_id, 'location': value})
+        log.debug('{id}: Setting location for {episode} to {location}',
+                  {'id': self.series.series_id, 'episode': self.identifier, 'location': value})
         self._location = value
-        self.file_size = os.path.getsize(value) if value and self.is_location_valid(value) else 0
+
+    @property
+    def file_size(self):
+        """Return the file size.
+
+        :return:
+        :rtype: int
+        """
+        return self._file_size
+
+    @file_size.setter
+    def file_size(self, size=None):
+        if size is not None:
```

Comment:
Do we really want to allow the optional `size`? 🤔

Reply:
Yeah I'm also curious in which situations where we do have a file, but don't have a size?